registration-processor/post-processor/registration-processor-printing-stage/src/main/java/io/mosip/registration/processor/print/stage/PrintingStage.java

```diff
@@ -343,17 +355,20 @@ public MessageDTO process(MessageDTO object) {
 		return object;
 	}
 
-	private CredentialRequestDto getCredentialRequestDto(String regId) {
+	private CredentialRequestDto getCredentialRequestDto(String regId,String issuerId,String credentialType,String template) {
 		CredentialRequestDto credentialRequestDto = new CredentialRequestDto();
+		Map<String, Object> additionalAttributes=new HashMap<>();
 
-		credentialRequestDto.setCredentialType(env.getProperty("mosip.registration.processor.credentialtype"));
+		credentialRequestDto.setCredentialType(credentialType);
 		credentialRequestDto.setEncrypt(encrypt);
 
 		credentialRequestDto.setId(regId);
 
-		credentialRequestDto.setIssuer(env.getProperty("mosip.registration.processor.issuer"));
+		credentialRequestDto.setIssuer(env.getProperty(issuerId));
 
 		credentialRequestDto.setEncryptionKey(generatePin());
+		additionalAttributes.put("template",template);
```

Comment:
I think this is not template itself, but template code, so please re-name the additional attribute name and variable accordingly.

Reply:
taken care